src/fmt.ml

```diff
@@ -65,6 +72,10 @@ let rec list ?(pp_sep = cut) pp_v ppf = function
 | v :: vs ->
     pp_v ppf v; if vs <> [] then (pp_sep ppf (); list ~pp_sep pp_v ppf vs)
 
+let hashtbl ?(pp_sep = cut) ~pp_k ~pp_v ppf tbl =
+  let f ppf k v = pf ppf "@[<1>%a:@.%a@]%a" pp_k k pp_v v pp_sep () in
```

Comment:
Looking at this again I'm a little bit unconvinced by what we concluded the other day. There's already a good deal of ad-hocness in that representation --- also why do you flush the pretty-printer on each binding ? That completely breaks the pretty printing process. 

If someone really wants to get it's own representation for a hashtable I think he's going to want full control. So I think I'm going to make a `hashtbl` combinator that is completely ad-hoc and maybe add another one `hashtbl_bindings` that pretty-prints the "list" of bindings using a combinator that formats the bindings (rather than separate key and value).